sqlglot/parser.py

```diff
@@ -7401,3 +7402,62 @@ def _parse_star_ops(self) -> exp.Star | exp.UnpackColumns:
                 "rename": self._parse_star_op("RENAME"),
             },
         )
+
+    def _parse_grant_privilage(self) -> t.Optional[exp.GrantPrivilage]:
+        privilage_parts = []
+
+        # Keep consuming consecutive keywords until comma (end of this privilage) or ON
+        # (end of privilage list) or L_PAREN (start of column list) are met
+        while self._curr and not self._match_set(
+            (TokenType.ON, TokenType.COMMA, TokenType.L_PAREN), advance=False
+        ):
+            privilage_parts.append(self._curr.text.upper())
+            self._advance()
+
+        this = exp.var(" ".join(privilage_parts))
+        expressions = (
+            self._parse_csv(self._parse_column) if self._match(TokenType.L_PAREN) else None
+        )
+        self._match(TokenType.R_PAREN)
```

Comment:
May be able to use `expressions = self._parse_wrapped_csv(..., optional=True)` here.

Reply:
On it's own it breaks tests as it consumes nearby privileges from what I see; We can still use it to save on the `R_PAREN` match line though